### PR TITLE
Feat/coordinator demo

### PR DESCRIPTION
**Pull request by veselaIl**

### CODE REVIEW

File: SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//  SFBaseKitDemo
+//
+//  Created by Vesela Ilchevska on 15.04.20.
+//  Copyright © 2020 Upnetix. All rights reserved.
+//
+
+import Foundation
+import SFBaseKit
+
+protocol LoginSceneDelegate: class {
```
Comment (dvp-petrov) [inline SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift]: This `class` might also be `Coordinator`

File: SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    func sceneShouldContinueToForgottenPassword()
+    func sceneShouldContinueToLoginIn()
+}
+
+class LoginCoordinator: Coordinator {
+    
+    // MARK: - Properties
+    private var navigationController: UINavigationController?
```
Comment (dvp-petrov) [inline SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift]: Having this in the `init` method, means that this property could be `let` and non-optional. Furthermore on this topic, you might want to make it `unowned` too for memory management purposes.

File: SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+//  Created by Vesela Ilchevska on 15.04.20.
+//  Copyright © 2020 Upnetix. All rights reserved.
+//
+
+import Foundation
+import SFBaseKit
+
+protocol LoginSceneDelegate: class {
+    func sceneShouldContinueToForgottenPassword()
+    func sceneShouldContinueToLoginIn()
```
Comment (dvp-petrov) [inline SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift]: Add the name of the scene in those methods.
If another scene also have continue to login method, the names will be the same - that is an issue for multi-protocol inheritance.
E.G. `loginSceneShouldContinueToLogIn()`

**Review by dvp-petrov — CHANGES_REQUESTED** [review]
Very good overall! 🙌 
Please consider presenting the forgot password screen and add HomeCoordinator after login.

File: SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift (hunk 1 above)
Comment (veselaIl) [reply SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift]: Changed :)

File: SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift (hunk 2 above)
Comment (veselaIl) [reply SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift]: Cool! Done

File: SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift (hunk 3 above)
Comment (veselaIl) [reply SFBaseKitDemo/SFBaseKitDemo/ExampleLoginModule/LoginCoordinator.swift]: Got that! Changed :)

**Review by aleksgeyman — APPROVED** [review]

**Review by dvp-petrov — APPROVED** [review]

**Pull request merged**